tests/acceptance/book-appointment-test.js

```diff
@@ -0,0 +1,445 @@
+    benificiary = {created_by_id: user.id, first_name: "Test", id: 12, identity_number: "1233",
+      identity_type_id: 1, last_name: "John", phone_number: "+85212312312", title: null};
+    identity_type1 = {id: 1, name: "Hong Kong Identity Card"};
+    package_type = make("package_type");
+    orderTransport = make("order_transport");
+    mocks = [];
+    $.mockjaxSettings.matchInRegistrationOrder = false;
+    mocks.push(
+      $.mockjax({url:`/api/v1/orders/*`,
+        type:'GET', status:200,
+        responseText: {
+          order: order.toJSON({includeId: true}),
+          goodcity_requests: [goodcityRequest]
+        }
+      }),
+      $.mockjax({url: "/api/v1/goodcity_request*", type: 'POST', status: 201,
+        responseText: {goodcity_request: goodcityRequest}}),
+      $.mockjax({url:"/api/v1/auth/current_user_profil*", responseText: user_profile }),
+      $.mockjax({url: "/api/v1/available_*", type: 'GET', status: 200,
+        responseText:["2018-06-14", "2018-06-15", "2018-06-16", "2018-06-19", "2018-06-20", "2018-06-21"]}),
+      mockFindAll("gogovan_transport").returns({json: {gogovan_transports: [gogo_van.toJSON({includeId: true})]}}),
+      mockFindAll('order').returns({ json: {orders: [order.toJSON({includeId: true})],
+        packages: [pkg.toJSON({includeId: true})], orders_packages: [ordersPackage.toJSON({includeId: true})]}}),
+      mockFindAll("booking_type").returns({json: {booking_types: [bookingType.toJSON({includeId: true})]}})
+    );
+    requestPurposeUrl = '/request_purpose';
+    clientInfoUrl = `/order/${order.id}/client_information`;
+    goodsDetailsUrl = `/order/${order.id}/goods_details`;
+    appointmentPageUrl = `/order/${order.id}/appointment_details`;
+    confirmBookingUrl = `/order/${order.id}/confirm_booking`;
+    bookingSuccessUrl = `/order/${order.id}/booking_success`;
+  },
+
+  afterEach: function() {
+     // Clear our ajax mocks
+    $.mockjaxSettings.matchInRegistrationOrder = true;
+    mocks.forEach($.mockjax.clear);
+
+    // Stop the app
+    Ember.run(App, 'destroy');
+  }
+});
+
+// home page login flow with book appointment test
+test("should redirect to login page if user not logged in", function(assert) {
+  window.localStorage.removeItem('authToken');
+  assert.expect(2);
+  visit('/');
+  andThen(function() {
+    click(".book-appointment");
+  });
+  andThen(function(){
+    assert.equal(currentURL(), '/login?bookAppointment=true');
+    assert.equal(Ember.$('.title').text().trim(), "Log in/Register");
+  });
+});
+
+// home page account details flow with book appointment test
+test("should redirect to Account details page if user is logged in and account details are incomplete", function(assert) {
+  assert.expect(2);
+  visit('/');
+  andThen(function() {
+    click(".book-appointment");
+  });
+  andThen(function(){
+    assert.equal(currentURL(), "/account_details?bookAppointment=true");
+    assert.equal(Ember.$('.title').text().trim(), "Account Details");
```

Comment:
Is it necessary to check ? I feel checking current url is fine ?

CC : @Darshan-Chauhan

Reply:
yeah, but i'm checking the UI also, so that, there's no mistake in UI also.

Reply:
Okay 👍